### PR TITLE
feat: replace `waitForRemotePeers()` with `waku.waitForPeer()` method

### PR DESCRIPTION
**Pull request by weboko**
## Problem

We have a common pattern when users absolutely need to:
```typescript
import { createLigthNode, waitForRemotePeers } from "@waku/sdk";

const waku = createLigthNode();

await waku.start();
await waitForRemotePeers(waku);
```

This PR makes `IWaku` self sufficient:
```typescript
import { createLigthNode } from "@waku/sdk";

const waku = createLigthNode();

await waku.start();
await waku.waitForPeer();
```

## Other stuff 

- Removed export of `waitForRemotePeers`;
- Renamed `Waku` interface to `IWaku`;
- Added nice docs:
image

## Notes

- Related to https://github.com/waku-org/js-waku/issues/1761

### CODE REVIEW

File: packages/interfaces/src/waku.ts [hunk 1]
```diff
@@ -10,36 +10,107 @@ import { Protocols } from "./protocols.js";
...
+   *  await waku.connect([Protocols.LightPush], 1000);
+   * } catch(e) {
+   *  waku.isConnected() === false;
+   *  console.error("Failed to connect due to", e);
+   * }
+   *
+   * waku.isConnected() === true;
+   * ```
+   */
+  connect(protocols?: Protocols[], timeoutMs?: number): Promise<void>;
```
Comment (weboko) [inline packages/interfaces/src/waku.ts:93]: @waku-org/js-waku looking for your feedback on naming

named it `connect` as when it is used it looks like `await waku.connect();` which seems convenient

Comment (weboko) [reply packages/interfaces/src/waku.ts]: agreed to use `.waitForPeer` instead 

@danisharora099 ping for your input

**Review by adklempner — APPROVED** [review]

File: packages/interfaces/src/relay.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ export interface IRelayAPI {
   readonly pubsubTopics: Set<PubsubTopic>;
   readonly gossipSub: GossipSub;
   start: () => Promise<void>;
+  waitForPeer: () => Promise<void>;
```
Comment (danisharora099) [inline packages/interfaces/src/relay.ts:19]:
```suggestion
  waitForPeers: () => Promise<void>;
```

File: packages/interfaces/src/waku.ts [hunk 3]
```diff
@@ -10,36 +10,107 @@ import { Protocols } from "./protocols.js";
...
+   * ```typescript
+   * await waku.stop();
+   *
+   * waku.isStarted === false;
+   * ```
+   */
   stop(): Promise<void>;
 
+  /**
+   * Resolves when Waku successfully gains connection to a remote peers that fits provided requirements.
```
Comment (danisharora099) [inline packages/interfaces/src/waku.ts:69]: let's either change this to `peer`, or change the flag to `peers`
in favour of the latter

```suggestion
   * Resolves when Waku successfully gains connection to a remote peer that fits provided requirements.
```

**Review by danisharora099 — COMMENTED** [review]
commented nits, lgtm

**Review by danisharora099 — APPROVED** [review]

File: packages/interfaces/src/waku.ts (hunk 1 above)
Comment (danisharora099) [reply packages/interfaces/src/waku.ts]: `waitForConnectivity` perhaps would be more verbose. Indifferent from now, we can improve this with external feedback

File: packages/interfaces/src/relay.ts (hunk 2 above)
Comment (weboko) [reply packages/interfaces/src/relay.ts]: indeed it is a good option, using it
cc @adklempner

**Pull request merged**